src/test/java/uk/nhs/ctp/auditFinder/AuditFinderTest.java

```diff
@@ -283,79 +237,72 @@ public void findAllEncounters_returnsAudits() throws IOException {
-    verify(elasticSearchClient).search(eq("test-supplier-audit"), searchSourceCaptor.capture());
-
-    var searchSource = searchSourceCaptor.getValue();
-
-    assertThat(searchSource.sorts(), hasSize(1));
-    assertThat(searchSource.sorts().get(0),
-        hasToString(equalToJSON("{ @timestamp : {order : asc } }")));
-    assertThat(searchSource.query(), hasToString(equalToJSON(
-        "{ bool : { must : ["
-            + " { term : { additionalProperties.supplierId : { value : test-supplier } } },"
-            + " { term : { additionalProperties.operation : { value : service_search } } }"
-            + "] } }")));
+  public void groupInteractions_withEmptyList_returnsEmpty() {
+    assertThat(auditFinder.groupInteractions(emptyList()), empty());
   }
 
   @Test
-  public void findAllServiceSearches_returnsAudits() throws IOException {
-    var auditSessionJson = "{ \"requestUrl\": \"/test-url\" }";
-    var auditSession = AuditSession.builder().requestUrl("/test-url").build();
-    var auditSessionWithEntryJson = "{"
-        + " \"requestUrl\": \"/test-url-2\","
-        + " \"entries\": ["
-        + " { \"requestUrl\": \"/test-url-3\" }"
-        + "] }";
-    var auditSessionWithEntry = AuditSession.builder()
-        .requestUrl("/test-url-2")
-        .entry(AuditEntry.builder().requestUrl("/test-url-3").build())
-        .build();
-
-    var searchHits = Stream.of(auditSessionJson, auditSessionWithEntryJson)
-        .map(ElasticSearchFixtures::buildSearchHit)
-        .collect(Collectors.toUnmodifiableList());
+  public void groupInteractions_withInvalidOperationType_fails() {
+    final var CREATION_DATE = Instant.parse("2019-08-22T12:11:54Z");
 
-    when(authService.requireSupplierId()).thenReturn("test-supplier");
-    when(elasticSearchClient.search(eq("test-supplier-audit"), any(SearchSourceBuilder.class)))
-        .thenReturn(searchHits);
-    when(objectMapper.readValue(auditSessionJson, AuditSession.class))
-        .thenReturn(auditSession);
-    when(objectMapper.readValue(auditSessionWithEntryJson, AuditSession.class))
-        .thenReturn(auditSessionWithEntry);
+    var encounter1Audit1 = AuditSession.builder()
+        .createdDate(CREATION_DATE)
+        .additionalProperty("operation", "invalid_operation_type")
+        .additionalProperty("interactionId", "1")
+        .build();
 
-    var audits = auditFinder.findAllServiceSearches();
+    var auditSessions = List.of(encounter1Audit1);
 
-    assertThat(audits, contains(auditSession, auditSessionWithEntry));
+    expectedException.expect(IllegalArgumentException.class);
+    auditFinder.groupInteractions(auditSessions);
   }
 
   @Test
-  public void objectMapper_readValue_canDeserialiseAudit() throws IOException {
-    var auditFile = getClass().getClassLoader().getResource("exampleAudit.json");
-    var auditJson = IOUtils.toString(Objects.requireNonNull(auditFile), StandardCharsets.UTF_8);
+  public void groupInteractions_returnsGroups() {
```

Comment:
could do with more tests here for:

- Downstream grouping by getCreatedDate
- Ordered by getStartedAt

Reply:
Added, see `AuditTransformerTest.java`